tests/webhook_test.go

```diff
@@ -350,5 +369,115 @@ var _ = Describe("Clone Auth Webhook tests", func() {
+
+				client, err := f.GetCdiClientForServiceAccount(targetNamespace.Name, serviceAccountName)
+				Expect(err).ToNot(HaveOccurred())
+
+				// can't list dvs in source
+				_, err = client.CdiV1beta1().DataVolumes(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{})
+				Expect(err).To(HaveOccurred())
+
+				// can list dvs in dest
+				_, err = client.CdiV1beta1().DataVolumes(targetNamespace.Name).List(context.TODO(), metav1.ListOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				// can't create clone of dv in source
+				_, err = client.CdiV1beta1().DataVolumes(targetNamespace.Name).Create(context.TODO(), targetDV, metav1.CreateOptions{})
+				Expect(err).To(HaveOccurred())
+
+				// let's do manual check as well
+				response, err := targetDV.AuthorizeSA(targetNamespace.Name, targetDV.Name, proxy, targetNamespace.Name, serviceAccountName)
+				fmt.Fprintf(GinkgoWriter, "response: %+v", response)
+				Expect(response.Allowed).To(BeFalse())
+				Expect(response.Reason).ToNot(BeEmpty())
+				Expect(err).ToNot(HaveOccurred())
+
+				addPermissionToNamespace(f.K8sClient, role, targetNamespace.Name, saName, groupName, f.Namespace.Name)
+				if !fail {
+					By("adding permissions on intermediate datasource")
+					addPermissionToNamespace(f.K8sClient, explicitRoleDataSource, targetNamespace.Name, saName, groupName, intermediateNamespace.Name)
+				}
+
+				// now can list dvs in source
+				Eventually(func() error {
+					_, err = client.CdiV1beta1().DataVolumes(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{})
+					return err
+				}, 60*time.Second, 2*time.Second).ShouldNot(HaveOccurred())
+
+				if fail {
+					// not sufficient
+					By("checking for insufficient permissions on intermediate datasource")
+					Consistently(func() error {
+						_, err = client.CdiV1beta1().DataVolumes(targetNamespace.Name).Create(context.TODO(), targetDV, metav1.CreateOptions{})
+						return err
+					}, 10*time.Second, 2*time.Second).Should(HaveOccurred())
+					Expect(err.Error()).To(ContainSubstring("insufficient permissions in intermediate datasource"))
+
+					return
+				}
+
+				By("checking sufficient permissions for pvc clone via intermediate datasource")
+				Eventually(func() error {
+					_, err = client.CdiV1beta1().DataVolumes(targetNamespace.Name).Create(context.TODO(), targetDV, metav1.CreateOptions{})
+					return err
+				}, 60*time.Second, 2*time.Second).ShouldNot(HaveOccurred())
+
+				// let's do another manual check as well
+				response, err = targetDV.AuthorizeSA(targetNamespace.Name, targetDV.Name, proxy, targetNamespace.Name, serviceAccountName)
+				Expect(response.Allowed).To(BeTrue())
+				Expect(response.Reason).To(BeEmpty())
+				Expect(err).ToNot(HaveOccurred())
+			},
+				Entry("when using explicit CDI permissions", explicitRole, serviceAccountName, "", false),
+				Entry("when using explicit CDI permissions without permissions on intermediate datasource", explicitRole, serviceAccountName, "", true),
```

Comment:
I would pass the role that makes this succeed as part of entry args instead of the !fail bit